### ISSUE TITLE
Support `--listen` for `tsh proxy` commands

### ISSUE THREAD

**Issue by greedy52** [gravitational/teleport#40509]
What would you like Teleport to do?
Ability to specify the listener address for the local proxies hosted by `tsh proxy` commands. (Maybe limit to local addresses only)

What problem does this solve?

Currently `tsh` local proxies only listen on `localhost` or `127.0.0.1`. Often applications are launched in docker containers so they cannot access these local proxies.

If a workaround exists, please include it.
`socat` ?

**Comment by greedy52** [comment]
There are internal discussions to use a unix socket `unix:/` which is a more secure option. Socket files can be shared locally to docker containers if needed. 

Apparently Machine ID already has this option:
- https://github.com/gravitational/teleport/pull/40992

**Comment by zzJinux** [comment]
Hi @Tener, I really needed this for at least `tsh proxy db` and built my own `tsh`. I've been using it for about 5 months without any issue. It allows specifying any `hostname:port` for the proxy address of `tsh proxy db`. I haven't added this feature to other commands yet.

Regarding contribution:
1. Is it ok to submit a PR for this?
2. Should I implement this for all proxy commands, or is `tsh proxy db` only enough?

**Comment by Tener** [comment]
@zzJinux feel free to raise a PR for just `tsh proxy db`. I don't expect the change to be complex, but care must be taken to properly warn about exposing unsecured TCP listeners; the security aspect is the tricky part.

Can you let us know the use case you are using this for?

**Comment by zzJinux** [comment]
@Tener it's a tiny change as you expected.

I manage multiple db connection configs in the TablePlus (GUI client). I wanted to automate the `tsh proxy db` setup when making a connection instead of manually spinning up one myself before every attempt. The bootstrap script feature is used to launch the `tsh proxy db` before the connection request is attempted. I must choose and reserve one address (like `127.0.0.1:5489`) per db connection config.

I prefer `127.0.0.2` over `127.0.0.1` for self-managed addresses because the port space of `127.0.0.1` is used by random programs, and it's tedious to control them not to collide with the ports I have reserved for self-managed use.

> the security aspect is the tricky part.

How about restricting the feature to only `127/8` addresses? Allowing only `127/8` does not satisfy what the OP requested, though 🤨.

**Comment by Tener** [comment]
Thank you, this is very useful. I like the subnet restriction; to achieve original goal we could have a flag that lifts it, like `--insecure-listen-anywhere`.

**Comment by zzJinux** [comment]
@Tener, I don't think this issue should be closed on PR-56003 merge. Can you prevent the auto-close?

**Comment by Tener** [comment]
> [@Tener](https://github.com/Tener), I don't think this issue should be closed on PR-56003 merge. Can you prevent the auto-close?

Should be good now.
